k8s/envoy-jwt/README.md

```diff
@@ -0,0 +1,329 @@
+   Selector      : k8s:container-name:auth-helper
+
+   Creating registration entry for the frontend - auth-server...
+   Entry ID      : 59a127fa-328c-4115-883e-5ee20b86714f
+   SPIFFE ID     : spiffe://example.org/ns/default/sa/default/frontend
+   Parent ID     : spiffe://example.org/ns/spire/sa/spire-agent
+   TTL           : 3600
+   Selector      : k8s:ns:default
+   Selector      : k8s:sa:default
+   Selector      : k8s:pod-label:app:frontend
+   Selector      : k8s:container-name:auth-helper
+   ...
+```
+
+Note that the selectors for new services points to the `auth-helper` container: `k8s:container-name:auth-helper`. This is necessary to authenticate the service into SPIRE so then it can fetch/validate the JWT SVIDs configured as an authentication header for every request.
+
+Intentionally, there is no registration entry for the frontend-2 service. It will be added later to demonstrate that requests are denied by the external authorization filter when a JWT-SVID is not present in the request header.
+
+
+# Part 2: Test Connections
+
+Now that services are deployed and also registered in SPIRE, let's test the authorization that we've configured.
+
+## Testing for Valid and Invalid JWT-SVIDs
+The first set of testing will demonstrate how valid JWT-SVIDs allow for the display of associated data and invalid JWT-SVIDs prevent the associated data from being displayed. To run these tests, we need to find the IP addresses and ports that make up the URLs to use for accessing the data.
+
+   ```console
+   $ kubectl get services
+
+   NAME            TYPE           CLUSTER-IP    EXTERNAL-IP      PORT(S)          AGE
+   backend-envoy   ClusterIP      None          <none>           9001/TCP         6m53s
+   frontend        LoadBalancer   10.8.14.117   35.222.164.221   3000:32586/TCP   6m52s
+   frontend-2      LoadBalancer   10.8.7.57     35.222.190.182   3002:32056/TCP   6m53s
+   kubernetes      ClusterIP      10.8.0.1      <none>           443/TCP          59m
+   ```
+
+The `frontend` service will be available at `EXTERNAL-IP` value and port `3000`, which was configured for our container. In the sample output shown above, the URL to navigate is `35.222.164.221:3000`. Open your browser and navigate to the IP address shown for `frontend` in your environment, adding the port `:3000`. Once the page is loaded, you'll see the account details for user _Jacob Marley_. 
+
+![Frontend][frontend-view]
+
+[frontend-view]: images/frontend_view.png "Frontend view"
+
+On the other hand, following the same steps we can confirm that the `frontend-2` service is not able to talk to the `backend` service. This is because it was not updated to include a JWT token in the request. The lack of a valid token on the request makes the Envoy instance in front of the `backend` reject it.
+As a result, when you connect to the URL for the `frontend-2` service (e.g. `35.222.190.182:3002`), the browser only displays the title without any account details.
```

Comment:
Let's put the browser test before the description of what's happening:
 
```suggestion
On the other hand, when you connect to the URL for the `frontend-2` service (e.g. `35.222.190.182:3002`), the browser only displays the title without any account details. This is because the `frontend-2` service was not updated to include a JWT token in the request. The lack of a valid token on the request makes the Envoy instance in front of the `backend` reject it.
```